pkgs/development/python-modules/nlpo3/default.nix

```diff
@@ -0,0 +1,64 @@
+{
+  lib,
+  buildPythonPackage,
+  rustPlatform,
+  fetchFromGitHub,
+  pythonOlder,
+  pythonAtLeast,
+
+  rustc,
+  cargo,
+
+  setuptools,
+  setuptools-rust,
+  wheel,
+}:
+
+buildPythonPackage rec {
+  pname = "nlpo3";
+  version = "1.3.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.6" || pythonAtLeast "3.12";
+
+  src = fetchFromGitHub {
+    owner = "PyThaiNLP";
+    repo = "nlpo3";
+    rev = "refs/tags/nlpo3-python-v${version}";
+    hash = "sha256-A1q6iieoPuRFTnyzg9y3Tp/rMx1c8nbcd1f8aaqonf4=";
+  };
+
+  sourceRoot = "${src.name}/nlpo3-python";
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    hash = "sha256-OWXD5t4Jg9/GLgwIvM1j4Ns5oFFaHWlZbGMh3cBsIQw=";
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.cargoSetupHook
+    rustc
+    cargo
+  ];
+
+  build-system = [
+    setuptools
+    setuptools-rust
+    wheel
+  ];
+
+  # need help with tests
+  # test failing:
+  # ModuleNotFoundError: No module named 'nlpo3._nlpo3_python_backend'
```

Comment:
I'd assume you've not packaged it correctly. It's here: https://github.com/PyThaiNLP/nlpo3/blob/658b9533385a01a8fe5a42bf37b0346c72afd93a/nlpo3-python/Cargo.toml#L11